numpy/core/code_generators/generate_umath_doc.py

```diff
@@ -0,0 +1,30 @@
+import sys
+import os
+import textwrap
+
+sys.path.insert(0, os.path.dirname(__file__))
+import ufunc_docstrings as docstrings
+sys.path.pop(0)
+
+def normalize_doc(docstring):
+    docstring = textwrap.dedent(docstring).strip()
+    docstring = docstring.encode('unicode-escape').decode('ascii')
+    docstring = docstring.replace(r'"', r'\"')
+    docstring = docstring.replace(r"'", r"\'")
+    # Split the docstring because some compilers (like MS) do not like big
+    # string literal in C code. We split at endlines because textwrap.wrap
+    # do not play well with \n
+    docstring = '\\n\"\"'.join(docstring.split(r"\n"))
+    return docstring
+
+def write_code(target):
+    with open(target, 'w') as fid:
+        fid.write(
+            "#ifndef _NPY_UMATH_DOCSTRING_H_\n"
```

Comment:
This is OK, but I have slowly been moving header guards to use the full path. Not sure it is worth the effort for a generated file.

Reply:
better to respect the new approach. I have replace it to the full path.